json_encoder_test.go

```diff
@@ -41,162 +41,154 @@ func assertJSON(t *testing.T, expected string, enc *jsonEncoder) {
 
 func withJSONEncoder(f func(*jsonEncoder)) {
 	enc := newJSONEncoder()
-	enc.AddString("foo", "bar")
 	f(enc)
 	enc.Free()
 }
 
-type noJSON struct{}
+type testBuffer struct {
+	bytes.Buffer
+}
 
-func (nj noJSON) MarshalJSON() ([]byte, error) {
-	return nil, errors.New("no")
+func (b *testBuffer) Sync() error {
+	return nil
 }
 
-func TestJSONAddString(t *testing.T) {
-	withJSONEncoder(func(enc *jsonEncoder) {
-		enc.AddString("baz", "bing")
-		assertJSON(t, `"foo":"bar","baz":"bing"`, enc)
+func (b *testBuffer) Lines() []string {
+	output := strings.Split(b.String(), "\n")
+	return output[:len(output)-1]
```

Comment:
Unfortunately, that doesn't behave the same way when the buffer is empty - splitting the empty string returns `[]string{""}`, which isn't what we want.